pkg/controller/restapi/operation/messaging/operation.go

```diff
@@ -0,0 +1,155 @@
+/*
+ *
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * /
+ *
+ */
+
+package messaging
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/controller/command"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/command/messaging"
+	resterrors "github.com/hyperledger/aries-framework-go/pkg/controller/restapi/errors"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/webhook"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
+	"github.com/hyperledger/aries-framework-go/pkg/internal/common/support"
+	"github.com/hyperledger/aries-framework-go/pkg/kms/legacykms"
+	"github.com/hyperledger/aries-framework-go/pkg/restapi/operation"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	// http over didcomm endpoints
+	httpOverDIDComm                = "/http-over-didcomm"
+	registerHTTPOverDIDCommService = httpOverDIDComm + "/register"
+
+	// message service endpoints
+	msgServiceOperationID = "/message"
+	registerMsgService    = msgServiceOperationID + "/register-service"
+	unregisterMsgService  = msgServiceOperationID + "/unregister-service"
+	msgServiceList        = msgServiceOperationID + "/services"
+	sendNewMsg            = msgServiceOperationID + "/send"
+	sendReplyMsg          = msgServiceOperationID + "/reply"
+)
+
+// provider contains dependencies for the common controller operations
+// and is typically created by using aries.Context()
+type provider interface {
+	OutboundDispatcher() dispatcher.Outbound
+	TransientStorageProvider() storage.Provider
+	StorageProvider() storage.Provider
+	LegacyKMS() legacykms.KeyManager
+}
+
+// Operation contains basic common operations provided by controller REST API
+type Operation struct {
+	command  *messaging.Command
+	handlers []operation.Handler
+}
+
+// New returns new common operations rest client instance
+func New(ctx provider, registrar operation.MessageHandler, notifier webhook.Notifier) (*Operation, error) {
+	msgcmd, err := messaging.New(ctx, registrar, notifier)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create messaging controller command: %w", err)
+	}
+
+	o := &Operation{command: msgcmd}
+
+	defer o.registerHandler()
```

Comment:
Why defer?

Reply:
we don't need it anymore, will remove it.